src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -460,8 +460,20 @@ private void verifyUserAccess(String jobId, AuthenticatedUserRequest userReq) {
       FlightMap inputParameters = flightState.getInputParameters();
       String flightSubjectId =
           inputParameters.get(JobMapKeys.SUBJECT_ID.getKeyName(), String.class);
+
       if (!StringUtils.equals(flightSubjectId, userReq.getSubjectId())) {
-        throw new JobUnauthorizedException("Unauthorized");
+        String resourceId =
```

Comment:
Yeah, I do think it would be cleaner to have this all contained in a single object.  It'd make this a bit more concise and less brittle (like if some job only had some of the fields and not all of them) I know we've had trouble with deserializing some objects in the past but I think this is vanilla enough that it should work

Reply:
I'll cosign this, I think it would make the relationship between these fields more clear.